packages/generate-pdf/src/index.ts

```diff
@@ -24,68 +23,136 @@ import {
   loadExternalFile,
 } from '@s1seven/schema-tools-utils';
 
-export { Content, StyleDictionary, TDocumentDefinitions, TFontDictionary } from 'pdfmake/interfaces';
+import { attachCertificateToPdf } from './attach-certificate-to-pdf';
+import { buildModule } from './build-module';
+import { makeA3Compliant } from './pdf-a-3a';
 
 export interface GeneratePdfOptions {
-  inputType?: 'json';
-  outputType?: 'buffer' | 'stream';
   generatorPath?: string;
   docDefinition?: Partial<TDocumentDefinitions>;
   fonts?: TFontDictionary;
   translations?: Translations;
   extraTranslations?: ExtraTranslations;
   languageFontMap?: LanguageFontMap;
+  attachCertificate?: boolean;
+  a3Compliant?: boolean;
+  title?: string;
 }
 
-export interface GeneratePdfOptionsExtended<T extends 'stream' | 'buffer'> extends GeneratePdfOptions {
-  outputType: T;
-}
-
-const fonts = {
-  Lato: {
-    normal: 'node_modules/lato-font/fonts/lato-normal/lato-normal.woff',
-    bold: 'node_modules/lato-font/fonts/lato-bold/lato-bold.woff',
-    italics: 'node_modules/lato-font/fonts/lato-light-italic/lato-light-italic.woff',
-    light: 'node_modules/lato-font/fonts/lato-light/lato-light.woff',
-  },
-};
-
-const generatePdfOptions: GeneratePdfOptions = {
-  inputType: 'json',
-  outputType: 'buffer',
-  fonts,
-};
-
 const baseDocDefinition = (content: TDocumentDefinitions['content']): TDocumentDefinitions => ({
   pageSize: 'A4',
   pageMargins: [20, 20, 20, 40],
   content,
   defaultStyle: {
-    font: 'Lato',
+    font: 'NotoSans',
   },
 });
 
-export async function buildModule(
-  filePath: string,
-  moduleName?: string,
-): Promise<{
-  //! TODO: create standard function signature!!!!
-  generateContent: (
-    certificate: Schemas,
-    translations: Translations,
-    languageFontMap?: LanguageFontMap,
-    extraTranslations?: ExtraTranslations,
-  ) => Content[];
-}> {
-  const code = await loadExternalFile(filePath, 'text');
-  const fileNamePrefix = moduleName || filePath;
-  const fileName = `${fileNamePrefix}-${new Date().toISOString()}`;
-  const _module = new Module(fileName);
-  _module.filename = fileName;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  (_module as any)._compile(code, fileName);
-  _module.loaded = true;
-  return _module.exports;
+const defaultFonts: TFontDictionary = {
+  NotoSans: {
+    normal: `${__dirname}/../../../fixtures/fonts/NotoSans-Regular.ttf`,
+    bold: `${__dirname}/../../../fixtures/fonts/NotoSans-Bold.ttf`,
+    italics: `${__dirname}/../../../fixtures/fonts/NotoSans-Italic.ttf`,
+  },
```

Comment:
Are you sure those fonts can be loaded when running the compiled code?

Reply:
We load fonts from the fixtures [in other places](https://github.com/s1seven/schema-tools/blob/main/utils/create-pdf-certificate.ts), but I am not sure how you make sure the fixtures are part of the output.